gapic/ads-templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2

```diff
@@ -918,7 +954,13 @@ def test_{{ method_name }}_pager(transport_name: str = "grpc"):
         {% endif %}
         pager = client.{{ method_name }}(request={})
 
-        assert pager._metadata == metadata
+        {% if service.version %}
+        if HAS_GOOGLE_API_CORE_VERSION_HEADER:
+            expected_metadata = tuple(expected_metadata) + (
```

Comment:
IIUC, `tuple + tuple == long-tuple`. So we have a flat list of header key/value pairs, correct? 

This seems brittle, since in general headers (maybe not the ones we're checking so far) can have more than one value. We should file an issue to be a bit more structured in our tests here. I also wonder about ordering (see my other comment).

Reply:
Filed https://github.com/googleapis/gapic-generator-python/issues/2024